indexer/cmd/indexer/main.go

```diff
@@ -12,8 +14,17 @@ var (
 )
 
 func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+
+	// Spinup a goroutine to catch interrupts and cancel the application context.
+	go func() {
+		opio.BlockOnInterrupts()
+		log.Crit("caught interrupt, shutting down...")
```

Comment:
log.Crit will panic. We don't use it except in rare code where we don't do error handling (and this code should do proper error handling).